shopify_python/google_styleguide.py

```diff
@@ -251,3 +256,45 @@ def __use_cond_expr(self, node):  # type: (astroid.If) -> None
                 else_body_name = node.orelse[0].targets[0].name
                 if if_body_name == else_body_name:
                     self.add_message('cond-expr', node=node)
+
+    def __lambda_func(self, node):  # type: (astroid.Lambda) -> None
+        """Prefer Operator Function to Lambda Functions"""
+
+        if isinstance(node.body, astroid.UnaryOp):
+            unary_operators = {
+                "~": "invert",
+                "-": "neg",
+                "not": "not_",
+                "+": "pos"
+            }
+            operator = unary_operators.get(node.body.op)
+            argname = node.args.args[0].name
+            if operator and not isinstance(node.body.operand, astroid.BinOp) and argname is node.body.operand.name:
+                varname = node.body.operand.name
+                lamfun = "lambda " + varname + ": " + node.body.op + " " + varname
+                opfun = "operator." + operator
+                self.add_message('lambda-func', node=node, args={'op': opfun, 'lamfun': lamfun})
+        elif isinstance(node.body, astroid.BinOp):
+            if shopify_python.ast.count_tree_size(node.body) == 3 and len(node.args.args) == 2:
+                node = node.body
+                binary_operators = {
+                    "+": "add",
+                    "-": "sub",
+                    "*": "mul",
+                    "/": "truediv",
+                    "**": "pow",
+                    "%": "modulo",
+                    "<": "lt",
+                    "<=": "le",
+                    "==": "eq",
+                    "!=": "ne",
+                    ">=": "ge",
+                    ">": "gt"
+                }
+                operator = binary_operators.get(node.op)
+                if operator:
+                    left = str(node.left.value) if node.left.name == 'int' else node.left.name
```

Comment:
I think if you verified that the operands were of the right node type you wouldn't have to worry about this case of literal operands, which I think you handle incorrectly anyway (see my requested test).